### ISSUE TITLE
`isValid` method?

### ISSUE THREAD

**Issue by niftylettuce** [jackbearheart/email-addresses#50]
Other packages such as `isemail`, `@hapi/address`, and `validator` seem to have validation bugs.  Parsing packages such as `addressparser` in `nodemailer/lib/addressparser` also seem to have issues when parsing certain addresses.  It would be useful if this package exposed a simple `isValid` method which returned `true` or `false`.  I'm assuming this could be a simple wrapper around `parseOneAddress` method which checked if the return value had a `typeof` with value of `"Object"` - but I haven't looked close enough yet to determine if that's the case.

**Comment by niftylettuce** [comment]
It looks like this is the case.  I don't write TypeScript and not sure if you would accept this PR.

**Comment by jackbearheart** [comment]
Yes, @niftylettuce , that is accurate. In your code, you could write `!!parseOneAddress(...)` to get this functionality.

I would accept such a PR! I would like to do two things: 1) add a function `parseOneMailbox` that is exactly like `parseOneAddress` except it looks for one `mailbox` instead of one `address` (ts return type `ParsedMailbox | null`) 2) add `function isMailbox(m) { return !!parseOneMailbox(m); }` (ts return type `boolean`). Happy to receive a PR like that or I'll get around to it someday.

**Comment by niftylettuce** [comment]
I don't write  nor support TypeScript so unfortunately cannot contribute

**Comment by xemlock** [comment]
@niftylettuce Watch out when using `parseOneAddress` for email validation, as current version (v3.1.0) can't properly handle invalid domains - see #53.
